autogpt_platform/frontend/src/components/agptui/PublishAgentSelect.tsx

```diff
@@ -0,0 +1,151 @@
+import * as React from "react";
+import Image from "next/image";
+
+interface Agent {
+  name: string;
+  lastEdited: string;
+  imageSrc: string;
+}
+
+interface PublishAgentSelectProps {
+  agents: Agent[];
+  onSelect: (agentName: string) => void;
+  onCancel: () => void;
+  onNext: () => void;
+  onClose: () => void;
+  onOpenBuilder: () => void;
+}
+
+export const PublishAgentSelect: React.FC<PublishAgentSelectProps> = ({
+  agents,
+  onSelect,
+  onCancel,
+  onNext,
+  onClose,
+  onOpenBuilder,
+}) => {
+  const [selectedAgent, setSelectedAgent] = React.useState<string | null>(null);
+
+  const handleAgentClick = (agentName: string) => {
+    setSelectedAgent(agentName);
+    onSelect(agentName);
+  };
+
+  return (
+    <div className="w-full max-w-[900px] bg-white rounded-3xl shadow-lg flex flex-col mx-auto">
+      <div className="p-4 sm:p-6 border-b border-slate-200 relative">
+        <div className="absolute top-4 right-4">
+          <button
+            onClick={onClose}
+            className="w-8 h-8 rounded-full bg-gray-100 flex items-center justify-center hover:bg-gray-200 transition-colors"
+            aria-label="Close"
+          >
+            <svg
```

Comment:
Can we add the svgs to components/ui/icons.tsx  in the same way the others have been. 

The is an icons.stories too so you can see what we have

Reply:
done this too